models/agent_helpers.go

```diff
@@ -120,12 +134,20 @@ func FindAgentsForNode(q *reform.Querier, nodeID string) ([]*Agent, error) {
 }
 
 // FindAgentsForService returns all Agents providing insights for given Service.
-func FindAgentsForService(q *reform.Querier, serviceID string) ([]*Agent, error) {
+func FindAgentsForService(q *reform.Querier, serviceID string, agentType *AgentType) ([]*Agent, error) {
 	if _, err := FindServiceByID(q, serviceID); err != nil {
 		return nil, err
 	}
 
-	structs, err := q.FindAllFrom(AgentTable, "service_id", serviceID)
+	conditions := []string{
+		"service_id = $1",
+	}
+	args := []interface{}{serviceID}
+	if agentType != nil {
+		conditions = append(conditions, "agent_type = $2")
+		args = append(args, agentType)
+	}
+	structs, err := q.SelectAllFrom(AgentTable, fmt.Sprintf("WHERE %s ORDER BY agent_id", strings.Join(conditions, " AND ")), args...)
```

Comment:
@BupycHuk, what is your opinion about to form this part of the query in go "text/template"?
IMHO it can be more readable.

Reply:
using text/template here is overengineering, IMHO